meta-mbl-distro/classes/mbl-partitions.bbclass

```diff
@@ -24,138 +24,407 @@ inherit mbl-artifact-names
-# Normal file system partitions, recorded in the partition table
+# WKS_BOOTLOADER1 (non-file system; single bank)
 # ------------------------------------------------------------------------------
-MBL_PARTITIONS = "\
-    ROOT \
-    BOOT \
-    BOOTFLAGS \
-    FACTORY_CONFIG \
-    NON_FACTORY_CONFIG \
-    LOG \
-    SCRATCH \
-    HOME \
-"
+# WKS_BOOTLOADER1, if it exists, has a single a non-file system partition
+# containing the first bootloader on the storage. It is typically TF-A BL2,
+# packaged however the vendor specific boot ROM requires.
+#
+MBL_WKS_BOOTLOADER1_FILENAME ?= "${MBL_BL2_FILENAME}"
+MBL_WKS_BOOTLOADER1_DEFAULT_SIZE_MiB = "3"
+MBL_WKS_BOOTLOADER1_ALIGN_KiB ?= "1"
+MBL_WKS_BOOTLOADER1_NO_DEFAULT_OFFSET = "1"
+MBL_WKS_BOOTLOADER1_SKIP ?= "0"
+MBL_WKS_BOOTLOADER1_NO_FS = "1"
+MBL_PARTITION_NAMES += "WKS_BOOTLOADER1"
 
-# For each <partition> in MBL_PARTITIONS, we define the following variables:
-# * MBL_<partition>_MOUNT_POINT # Mountpoint for partition
-# * MBL_<partition>_LABEL # Default label of partition (without any suffix for the bank)
-# * MBL_<partition>_SIZE_MiB # Default size of partition in Mebibytes
-# * MBL_<partition>_ALIGN_KiB # Default alignment of partition in Kibibytes
+# ------------------------------------------------------------------------------
+# WKS_BOOTLOADER2 (non-file system; two banks)
+# ------------------------------------------------------------------------------
+# WKS_BOOTLOADER2, if it exists, has a banked pair of non-file system
+# partitions containing the next bootloader on the storage. It is Typically a
+# FIP image containing BL3, OP-TEE and U-Boot.
 #
+MBL_WKS_BOOTLOADER2_FILENAME ?= "${MBL_FIP_BIN_FILENAME}"
+MBL_WKS_BOOTLOADER2_DEFAULT_SIZE_MiB = "4"
+MBL_WKS_BOOTLOADER2_IS_BANKED = "1"
+MBL_WKS_BOOTLOADER2_SKIP ?= "0"
+MBL_WKS_BOOTLOADER2_NO_FS = "1"
+MBL_PARTITION_NAMES += "WKS_BOOTLOADER2"
+
 # ------------------------------------------------------------------------------
-# Root partition (two banks)
+# WKS_BOOTLOADER_FS (file system; single bank)
 # ------------------------------------------------------------------------------
-MBL_ROOT_MOUNT_POINT = "/"
-MBL_ROOT_PARTITION_LABEL ?= "rootfs"
-MBL_ROOT_PARTITION_SIZE_MiB ?= "500"
-MBL_ROOT_PARTITION_ALIGN_KiB ?= "${MBL_FLASH_ERASE_BLOCK_SIZE_KiB}"
+# WKS_BOOTLOADER_FS, if it exists, has a single file system partition and can
+# be used to store a bootloader on e.g. Raspberry Pi where the boot ROM expects
+# to find a bootloader on a file system partition.
+# 
+MBL_WKS_BOOTLOADER_FS_MOUNT_POINT = "${MBL_BOOTLOADER_FS_DIR}"
+MBL_WKS_BOOTLOADER_FS_LABEL ?= "bootloaderfs"
+MBL_WKS_BOOTLOADER_FS_DEFAULT_SIZE_MiB ?= "10"
+MBL_WKS_BOOTLOADER_FS_SKIP ?= "1"
+MBL_WKS_BOOTLOADER_FS_FSTYPE ?= "vfat"
+MBL_WKS_BOOTLOADER_FS_MOUNT_OPTS ?= "ro,defaults"
+MBL_PARTITION_NAMES += "WKS_BOOTLOADER_FS"
 
 # ------------------------------------------------------------------------------
-# Boot partition
+# BOOT (file system; single bank)
```

Comment:
> Should we mention that this is the partition that contains the kernel, device-tree and initramfs?

Will do.

> Why the BOOT partition is not banked?

Just because I haven't done it yet... I'm thinking that I should probably have a separate PR for:
* Adding the WKS_BOOTLOADER_FS partition.
* Banking the BOOT partition.

Actually using the WKS_BOOTLOADER_FS partition for rpi3 won't work without fiddling with our U-Boot scripts and I don't want to make this PR too big.

Reply:
> Should we mention that this is the partition that contains the kernel, device-tree and initramfs?

https://github.com/ARMmbed/meta-mbl/pull/617/commits/34e632c489ddce4f816e5f19992c7b47b7c6432a

Reply:
> I'm thinking that I should probably have a separate PR for...Adding the WKS_BOOTLOADER_FS partition.

https://github.com/ARMmbed/meta-mbl/pull/617/commits/25e076d83d21610aa4390757cc5182af04ff9521, https://github.com/ARMmbed/meta-mbl/pull/617/commits/478648f56ea6c6fe314c11e8639efd3f56955eeb